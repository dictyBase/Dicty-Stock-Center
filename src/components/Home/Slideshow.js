// @flow
import React from "react"
import { Carousel } from "react-responsive-carousel"
import ctr9 from "static/slideshow-images/ctr9-mutant-DG1071.png"
import gbqA from "static/slideshow-images/gbqA-mutant-DG1120.png"
import ggtA from "static/slideshow-images/ggtA-mutant-DG1109.png"
import tipB from "static/slideshow-images/tipB-mutant-DG1036.png"
import "react-responsive-carousel/lib/styles/carousel.min.css"

/**
 * Slideshow is an image slideshow carousel with curated dicty photos.
 */

const Slideshow = () => (
  <Carousel
    showStatus={false}
    showIndicators={false}
    showThumbs={false}
    autoPlay
    interval={5000}
    infiniteLoop>
    <div>
      <img src={ctr9} alt="ctr9-mutant-DG1071" />
    </div>
    <div>
      <img src={gbqA} alt="gbqA-mutant-DG1120" />
    </div>
    <div>
      <img src={ggtA} alt="ggtA-mutant-DG1109" />
    </div>
    <div>
      <img src={tipB} alt="tipB-mutant-DG1036" />
    </div>
  </Carousel>
)

export default Slideshow
