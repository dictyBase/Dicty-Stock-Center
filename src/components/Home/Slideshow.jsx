import React, { Component } from 'react'
import { Carousel } from 'react-responsive-carousel'
import 'react-responsive-carousel/lib/styles/carousel.min.css'

export default class Slideshow extends Component {
  displayName = 'front page carousel'

  render() {
      return (
      <Carousel
        showStatus={ false }
        showIndicators={ false }
        showThumbs={ false }
        autoPlay
        interval={ 5000 }
        infiniteLoop>
        <div>
          <img src="https://i.imgur.com/SqHpamt.png" alt="" />
        </div>
        <div>
          <img src="https://i.imgur.com/gP5FAAf.png" alt="" />
        </div>
        <div>
          <img src="https://i.imgur.com/pgX6ZhN.png" alt="" />
        </div>
        <div>
          <img src="https://i.imgur.com/6srEh3F.png" alt="" />
        </div>
        <div>
          <img src="http://betatest.dictybase.org/images/dictyFront01.jpg" alt="" />
        </div>
      </Carousel>
    )
  }
}
