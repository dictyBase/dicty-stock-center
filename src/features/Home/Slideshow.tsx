import React from "react"
import { Carousel } from "react-responsive-carousel"
import { makeStyles } from "@material-ui/core/styles"
import ctr9 from "common/assets/slideshow-images/ctr9-mutant-DG1071.png"
import gbqA from "common/assets/slideshow-images/gbqA-mutant-DG1120.png"
import ggtA from "common/assets/slideshow-images/ggtA-mutant-DG1109.png"
import tipB from "common/assets/slideshow-images/tipB-mutant-DG1036.png"
import "react-responsive-carousel/lib/styles/carousel.min.css"

const useStyles = makeStyles({
  container: {
    marginBottom: "24px",
  },
})

/**
 * Slideshow is an image slideshow carousel with curated dicty photos.
 */

const Slideshow = () => {
  const classes = useStyles()

  return (
    <Carousel
      className={classes.container}
      showStatus={false}
      showIndicators={false}
      showThumbs={false}
      autoPlay
      interval={5000}
      infiniteLoop>
      <div>
        <img src={ctr9} alt="ctr9-mutant-DG1071" />
      </div>
      <div>
        <img src={gbqA} alt="gbqA-mutant-DG1120" />
      </div>
      <div>
        <img src={ggtA} alt="ggtA-mutant-DG1109" />
      </div>
      <div>
        <img src={tipB} alt="tipB-mutant-DG1036" />
      </div>
    </Carousel>
  )
}

export default Slideshow
