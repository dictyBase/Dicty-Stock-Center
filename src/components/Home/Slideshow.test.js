import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Slideshow from './Slideshow'

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Slideshow />, div)
})

test('matching a snapshot of Carousel', () => {
    const component = renderer.create(<Slideshow />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
