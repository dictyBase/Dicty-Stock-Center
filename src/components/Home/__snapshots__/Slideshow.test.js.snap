// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Carousel 1`] = `
<div
  className={undefined}
>
  <div
    className="carousel carousel-slider"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <button
      className="control-arrow control-prev"
      onClick={[Function]}
      type="button"
    />
    <div
      className="slider-wrapper axis-horizontal"
      style={Object {}}
    >
      <ul
        className="slider animated"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "MozTransform": "translate3d(0%,0,0)",
            "MozTransitionDuration": "350ms",
            "MsTransform": "translate3d(0%,0,0)",
            "MsTransitionDuration": "350ms",
            "OTransform": "translate3d(0%,0,0)",
            "OTransitionDuration": "350ms",
            "WebkitTransform": "translate3d(0%,0,0)",
            "WebkitTransitionDuration": "350ms",
            "msTransform": "translate3d(0%,0,0)",
            "msTransitionDuration": "350ms",
            "transform": "translate3d(0%,0,0)",
            "transitionDuration": "350ms",
          }
        }
      >
        <li
          className="slide selected"
          onClick={[Function]}
        >
          <div>
            <img
              src="https://i.imgur.com/SqHpamt.png"
            />
          </div>
        </li>
        <li
          className="slide"
          onClick={[Function]}
        >
          <div>
            <img
              src="https://i.imgur.com/gP5FAAf.png"
            />
          </div>
        </li>
        <li
          className="slide"
          onClick={[Function]}
        >
          <div>
            <img
              src="https://i.imgur.com/pgX6ZhN.png"
            />
          </div>
        </li>
        <li
          className="slide"
          onClick={[Function]}
        >
          <div>
            <img
              src="https://i.imgur.com/6srEh3F.png"
            />
          </div>
        </li>
        <li
          className="slide"
          onClick={[Function]}
        >
          <div>
            <img
              src="http://betatest.dictybase.org/images/dictyFront01.jpg"
            />
          </div>
        </li>
      </ul>
    </div>
    <button
      className="control-arrow control-next"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
